controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -99,11 +137,414 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases(ctx context.Contex
+				r.log.Error(err, "Failed to update StorageClassClaim with finalizer.", "StorageClassClaim", klog.KRef(r.storageClassClaim.Namespace, r.storageClassClaim.Name))
+				return reconcile.Result{}, err
+			}
+		}
+		if r.storageClassClaim.Spec.Type == "blockpool" {
+			if err := r.reconcileCephClientRBDProvisioner(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephClientRBDNode(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephBlockPool(storagecluster); err != nil {
+				return reconcile.Result{}, err
+			}
+		} else if r.storageClassClaim.Spec.Type == "sharedfilesystem" {
+			if err := r.reconcileCephClientCephFSProvisioner(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephClientCephFSNode(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephFilesystemSubVolumeGroup(); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		cephResourcesReady := true
+		for _, cephResource := range r.storageClassClaim.Status.CephResources {
+			if cephResource.Phase != "Ready" {
+				cephResourcesReady = false
+				break
+			}
+		}
+
+		if cephResourcesReady {
+			r.storageClassClaim.Status.Phase = string(v1alpha1.StorageClassClaimStateReady)
+		}
+
+	} else {
+		// todo deletion of the Ceph Resources
+	}
+
+	r.log.Info("Running StorageClassClaim controller in Converged/Provider Mode")
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClassClaimReconciler) initReconciler(request reconcile.Request) {
+	r.namespace = request.Namespace
+	ownerRef := metav1.OwnerReference{
+		UID:        r.storageClassClaim.UID,
+		APIVersion: r.storageClassClaim.APIVersion,
+		Kind:       r.storageClassClaim.Kind,
+		Name:       r.storageClassClaim.Name,
+	}
+	if r.storageClassClaim.Spec.Type == "cephBlockPool" {
+		r.cephBlockPool = &rookCephv1.CephBlockPool{}
+		r.cephBlockPool.Name = fmt.Sprintf("%s-%s", "cephblockpool", r.storageClassClaim.Name)
+		r.cephBlockPool.Namespace = r.namespace
+		r.cephBlockPool.OwnerReferences = []metav1.OwnerReference{ownerRef}
+
+		r.cephClientRBDProvisioner = &rookCephv1.CephClient{}
+		r.cephClientRBDProvisioner.Name = controllers.GenerateHashForCephClient(r.storageClassClaim.Name, "rbd", "provisioner")
+		r.cephClientRBDProvisioner.Namespace = r.namespace
+		r.cephClientRBDProvisioner.OwnerReferences = []metav1.OwnerReference{ownerRef}
+
+		r.cephClientRBDNode = &rookCephv1.CephClient{}
+		r.cephClientRBDNode.Name = controllers.GenerateHashForCephClient(r.storageClassClaim.Name, "rbd", "node")
+		r.cephClientRBDNode.Namespace = r.namespace
+		r.cephClientRBDNode.OwnerReferences = []metav1.OwnerReference{ownerRef}
+	} else if r.storageClassClaim.Spec.Type == "cephFileSystemSubVolumeGroup" {
+		r.cephFilesystemSubVolumeGroup = &rookCephv1.CephFilesystemSubVolumeGroup{}
+		r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("%s-%s", "cephfilesystemsubvolumegroup", r.storageClassClaim.Name)
```

Comment:
> If you have a constat part you can always bake it into the string

ack

> 
> But regardless, this might be problematic in cases where the claim name is already at the max length for a k8s resource, we already hit these kinds of issues in the past.

what are you suggesting here? we keep it for now or some other naming